app/src/main/java/com/vci/vectorcamapp/core/presentation/util/error/ErrorSnackbarHost.kt

```diff
@@ -0,0 +1,65 @@
+package com.vci.vectorcamapp.core.presentation.util.error
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Snackbar
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.flow.SharedFlow
+
+@Composable
+fun ErrorSnackbarHost(
+    modifier: Modifier = Modifier, errorFlow: SharedFlow<ErrorData> = LocalErrorDataFlow.current
+) {
+    val snackbarHostState = remember { SnackbarHostState() }
+    val context = LocalContext.current
+
+    LaunchedEffect(errorFlow) {
+        errorFlow.collect { errorData ->
+            snackbarHostState.currentSnackbarData?.dismiss()
```

Comment:
We have an ObserveAsEvents function which can deal with flows. Would that work here?